### ISSUE TITLE
Make it easier to create the VMs on a non-standard location

### ISSUE THREAD

**Issue by yosuah** [docker/toolbox#191]
Hi,

Currently it is a non-trivial issue to create or move the created VM on a new location. This is important because by default Toolbox creates a VirtualBox VM under the curretn user's home dir, which is normally on drive C;. As many people have a smaller SSD and a larger secondary drive, it is likely that they want to move Docker to the secondary drive either during installation or sometimes later.

I faced the same issue and I had to take the following steps to move my VM. I may have overcomplicated some steps, but it took quite some time to figure everrything out.

* Download and install Docker Toolbox
* Copy the VM to a different drive if you are running out of disk space.
  * Stop the machine in VBox
  * Remove the disk under File -> Virtual media manager
  * Remove the machine but do not delete the files
  * Copy the whole [userdir]\.docker\machines directory to the new drive
  * Add the machine using Machine -> Add (not File -> Import appliance)
  * Add/correct the disk path under Settings -> Storage
  * Update the docker startes script: edit C:\Program Files\Docker Toolbox\start.sh and add this in the beginning (e.g. at line 7):
    * export MACHINE_STORAGE_PATH="F:\VirtualBox VMs\docker-machine"
    * Make sure that there is no trailing \ at the end of the path
  * Update the Cert paths for the command
    * I tried setting these in start.sh, but they didn’t work:
      * export MACHINE_TLS_CA_CERT="F:\VirtualBox VMs\docker-machine\machines\default\ca.pem"
      * export MACHINE_TLS_CA_KEY="F:\VirtualBox VMs\docker-machine\machines\default\cert.pem"
      * export MACHINE_TLS_CLIENT_CERT="F:\VirtualBox VMs\docker-machine\machines\default\server.pem"
      * export MACHINE_TLS_CLIENT_KEY="F:\VirtualBox VMs\docker-machine\machines\default\server-key.pem"
    * I ended up copying back the certs to the original location..

This doesn't even include the additional steps you need to do (e.g. change file sharing between the VM and the host, set up port forwarding, etc.). Having to do so much before you can start using your first container is a bit disengaging for new (and existing) users. I would recommend adding options for setting these things up during the first installation of Toolbox if at all possible.

Cheers,
Adam

**Comment by mchiang0610** [comment]
@yosuah Thanks for this.

**Comment by ChaosD** [comment]
Subscribing, iam in the same boat.

**Comment by pyrotek** [comment]
I couldn't agree more. My SSD drive is only 80gbs in size, 80% of which is filled by Windows and critical system tools. I really need docker to stay confined to my HDD, which is on my system for this purpose. Looks like I'm going to have to do it the long way :tired_face:

Thanks for sharing the info!

**Comment by potz** [comment]
Thank you for your very thorough explanation. I second this request, SSDs are normally small and I'm sure lot of people are feeling the same pain.

**Comment by tsmgodoi** [comment]
One possible solution is to create a directory junction as follows:
1) Copy the folder <USERDIR>\.docker to F:\Docker\.docker
2) Delete the folder <USERDIR>\.docker\machines
3) Create a junction with: mklink /j <USERDIR>\.docker\machine F:\Docker\.docker\machine

**Comment by potz** [comment]
@tsmgodoi thanks for great workaround idea :+1:

**Comment by juliangamble** [comment]
Try this one: http://stackoverflow.com/a/33687993/15441

**Comment by weeblr** [comment]
Hi
I have had good success selecting the storage path by changing the start.sh file as in this PR: #452 
After that change, it only takes setting the MACHINE_STORAGE_PATH to the desired folder.

Cheers

**Comment by ArunRajV4444** [comment]
Thank you @tsmgodoi . for me Kitematic was not working after I set the MACHINE_STORAGE_PATH to some other folder. Your simple solution for creating a .docker directory junction did the trick for me. Only change is that I had to set the MACHINE_STORAGE_PATH=<new location>/.docker/machine.

**Comment by putztzu** [comment]
Agree with comments in this thread.
An additional modification to the comment by weeblr, 

I already had VBox and Github on Windows installed, I was wary that the Toolbox install might wipe out my various custom settings so excluded them during the install. But, apparently Toolbox fundamentally does not build on top of VBox configuration and instead implements its own which I believe is the fundamental install and running logic error. Toolbox should be designed as a shell on top of the components it orchestrates. In other words, VBox configurations should reside in VBox. Git configurations should reside in Git. And, so on. Then Toolbox can sit on top of everything with its own configuration. 

An example is this MACHINE_STORAGE_PATH. 
This should not be specified. Instead, the install should simply create the Guest using whatever values are stored in VBox.

By disconnecting Toolbox from its underlying components, this would likely also address a number of other feature requests like supporting other virtualization than VBox,

**Comment by weeblr** [comment]
Hi

>  VBox configurations should reside in VBox.

> An example is this MACHINE_STORAGE_PATH. 
> This should not be specified. Instead, the install should simply create the Guest using whatever values are stored in VBox.

@putztzu : We are not talking about Vbox configuration here. VBox only has a config for the **default** location of VMs. We are indeed talking about a Toolbox configuration: where does Toolbox should store the VM it creates through VM.
Rgds

**Comment by micheljung** [comment]
Sounds like a low hanging fruit to me. Would be very much appreciated!

**Comment by GameKyuubi** [comment]
Yeah this is a huge pain in the neck that wouldn't take much work to fix.
